File: java/src/main/java/SimpleBookPricer.java
public class SimpleBookPricer {

    public static double calculatePrice(String string, String string2, Book.Type type, double rrp, int i) {
        
        if(Book.Type.PAPERBACK == type){
            return rrp;
        }
    
        if(i > 6){
            double d = rrp / 2d;
            return Math.round(d);
        }

        var discount = 0;

        if(i == 0){
            discount = 2;
        }

        return rrp - discount;
    }

}

File: typescript/src/SimpleBookPricer.ts
import { Book, BookType } from "./Book";


export class SimpleBookPricer {
    public static calculatePrice(
        title: string,
        author: string,
        type: BookType,
        rrp: number,
        month: number
    ): number {
        if (type === BookType.PAPERBACK) {
            return rrp;
        }

        if (month > 6) {
            const discountedPrice = rrp / 2;
            return Math.round(discountedPrice);
        }

        let discount = 0;

        if (month === 0) {
            discount = 2;
        }

        return rrp - discount;
    }
}
